app/controllers/users_controller.rb

```diff
@@ -0,0 +1,21 @@
+class UsersController < ApplicationController
+  def create
+    user = User.new(permitted_params)
+    if user.save
+      render json: user, status: ok
+    else
+      render json: user.errors, status: unprocessable_entity
+    end
+  end
+
+  def index
+    users_posts = User.includes(:posts)
+    render json: users_posts.to_json(include: :posts)
```

Comment:
This is unnecessary. use serializers to achieve this

Reply:
Okay will do it using serializer